File: 324. Wiggle Sort II.py
class Solution(object):
    def wiggleSort(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        temp = sorted(nums)
        s = (len(nums)+1)/2
        l = len(nums)
        for i in xrange(len(nums)):
            if i & 1 == 0:
                s -= 1
                nums[i] = temp[s]
            else:
                l -= 1
                nums[i] = temp[l]

File: 324. Wiggle Sort II.cpp
class Solution {
public:
    void wiggleSort(vector<int>& nums) {
        vector<int> temp = nums;
        sort(temp.begin(), temp.end());
        int s = (int) ((nums.size() + 1) / 2);
        int l = nums.size();
        for (int i = 0; i < nums.size(); i++)
        {
            if (i % 2 == 0)
            {
                s--;
                nums[i] = temp[s];
            }
            else
            {
                l--;
                nums[i] = temp[l];
            }
        }
    }
};
